### DISCUSSION TITLE
Can someone explain how exactly does the "LoadThenNavigate" case study work?

### DISCUSSION THREAD

**Discussion by phi161** [pointfreeco/swift-composable-architecture#1414] [Q&A] [answered]
I am studying the [LoadThenNavigate](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-LoadThenNavigate.swift) example and even if I can follow what happens _after_ `LoadThenNavigateAction.setNavigation(isActive: true)` is sent, I can't understand _who_ sends this action.

Does tapping on the row cause the `isActive` [binding](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-LoadThenNavigate.swift#L90-L93) to send the `LoadThenNavigateAction.setNavigation` action? And if yes, why is its `isActive` associated value set to `true`? I would expect it to be `false`, because `isNavigationActive` is also `false`.

**Comment by mbrandonw** [comment, marked as answer]
That case study demonstrates how to execute some logic before causing a drill-down navigation. Like if you wanted to perform an API request and then with the result of that request decide whether you do the drill-down or show an error message.

So when you tap the nav link, a `setNavigation(true)` action is sent, but we don't actually update state to have `isNavigationActive=true`. Instead, we fire off an effect, and when the effect finishes we then set `isNavigationActive=true` which then causes the drill-down to happen.

Does that help?

**Reply by phi161** [reply]
> So when you tap the nav link, a `setNavigation(true)` action is sent

This is the part I don't understand - the `isActive` binding looks like this:

```
viewStore.binding(
            get: \.isNavigationActive,
            send: LoadThenNavigateAction.setNavigation(isActive:)
          )
```

How come and tapping on the link sends `setNavigation(true)`?

**Reply by mbrandonw** [reply]
That is just how SwiftUI works. The boolean binding you hand to `NavigationLink.init(isActive)` gets `true` written to it when it is tapped. We then intercept that `true` and send an action to the view instead of writing state directly, which gives us the opportunity to layer on logic before actually performing the navigation.

**Reply by phi161** [reply]
Perfect, I get it now - thanks a lot for the quick replies!
